lib/task_helpers.rb

```diff
@@ -1,213 +1,106 @@
-        index_name = create_index()
-        # WARNING: if an index exists with the same name as the intended alias, it
-        #   will be deleted.
-        move_alias(alias_name, index_name, force_delete: true)
+    def self.initialize_indices(force_new_index)
+      # When force_new_index is true, fresh indices will be created even if it already exists.
+      if force_new_index or not exists_indices
+        create_indices
       else
-        LOG.info "Skipping initialization. The 'content' alias already exists. If 'rake search:validate_index' indicates "\
-          "a problem with the mappings, you could either use 'rake search:rebuild_index' to reload from the db or 'rake "\
+        LOG.info "Skipping initialization. Indices already exist. If 'rake search:validate_indices' indicates "\
+          "a problem with the mappings, you could either use 'rake search:rebuild_indices' to reload from the db or 'rake "\
           "search:initialize[true]' to force initialization with an empty index."
       end
     end
 
-    def self.put_mappings(name)
-      # As of ES 0.9, the order that these mappings are created matters.  Unit test failures
-      # appear with a different order. It is unclear if this is a defect in ES, the test, or
-      # neither.
-      [CommentThread, Comment].each do |model|
-        Elasticsearch::Model.client.indices.put_mapping(index: name, type: model.document_type, body: model.mappings.to_hash)
-      end
-      LOG.info "Added mappings to index: #{name}."
-    end
+    # Validates that each index includes the proper mappings.
+    # There is no return value, but an exception is raised if the index is invalid.
+    def self.validate_indices
 
-    # Validates that the alias exists and its index includes the proper mappings.
-    # There is no return value, but an exception is raised if the alias is invalid.
-    #
-    # Params:
-    # +alias_name+:: The alias name to be validated.
-    def self.validate_index(alias_name)
-      if exists_alias(alias_name) === false
-        fail "Alias '#{alias_name}' does not exist."
-      end
-
-      actual_mapping = Elasticsearch::Model.client.indices.get_mapping(index: alias_name).values[0]['mappings']
-      expected_mapping = {}
-      [CommentThread, Comment].each do |model|
-        expected_mapping.merge! model.mappings.to_hash
-      end
+      actual_mappings = Elasticsearch::Model.client.indices.get_mapping(index: INDEX_NAMES)
 
-      # As of ES 0.9, the order the mappings are created in matters.  See put_mappings.
-      # Compare document types and order
-      expected_mapping_keys = expected_mapping.keys.map { |x| x.to_s }
-      if actual_mapping.keys != expected_mapping_keys
-        fail "Actual mapping types [#{actual_mapping.keys}] does not match expected mapping types (including order) [#{expected_mapping.keys}]."
-      end
-
-      # Check that expected field mappings of the correct type exist
-      expected_mapping.keys.each do |doc_type|
```

Comment:
My impression here is that we're getting rid of this detailed report of what's different between the mappings and just letting the operator figure it out themselves?  (That's probably fine, just want to make sure I understand why all this code seems to be gone...)

Reply:
I returned this part of code, thanks. I miss that it check index mapping structure. Done